### PR TITLE
WT-8631 query_timestamp names should match set_timestamp names

### PR DESCRIPTION
**Pull request by keithbostic**
Use "oldest_timestamp/stable_timestamp" instead of "oldest/stable" so the names used by `WT_CONNECTION.query_timestamp` match the names used by `WT_CONNECTION.set_timestamp`.

@agorrod, @michaelcahill, @bvpvamsikrishna: the reason for this ticket that when writing new timestamp documentation, I found it was difficult to distinguish between "oldest" and "stable" (as arguments to `query_timestamp`), and phrases like "the stable timestamp" or "the oldest timestamp". It also wasn't as obvious as I'd like that `query_timestamp` using those strings returned **differently** named global timestamp values, especially since other query arguments return manufactured or aggregated values, as opposed to single, specific values.

I'm not going to fight hard for this one, and I completely understand if my personal preferences differ from everyone else's, but I believe it's easier to explain how timestamps work if we use the same names for `query_timestamp` as for `set_timestamp`.

@tammybailey, I don't think the import code has made it out into the wild, so I didn't make the import configuration strings backward compatible, I just switched them to the new versions. Please stop me if that's not the case.

### CODE REVIEW

**Comment by sueloverso** [comment]
I am pretty sure the import code is used by MongoDB. Whether they use the specific configuration setting you changed I don't know.

**Comment by keithbostic** [comment]
> I am pretty sure the import code is used by MongoDB. Whether they use the specific configuration setting you changed I don't know.

Thanks. It’s easy enough to support both old and new configs.

**Comment by tammybailey** [comment]
@keithbostic The import code is indeed used by MongoDB. The  timestamp configuration options are used as well (by the serverless merge effort, for example).

**Comment by keithbostic** [comment]
@tammybailey, @sueloverso: I've pushed backward compatible support for import in cf28a78.

**Comment by keithbostic** [comment]
@tammybailey, I'm probably missing something obvious, but I want to be sure: I don't see anywhere we check an import load for `compare_timestamp=(oldest_timestamp)`, and I don't see a test path for it. Is `oldest_timestamp` the default or something so it's not necessary?

**Comment by tammybailey** [comment]
@quchenhao can you answer Keith's question above? I believe this is code that was added as part of the merge project.

**Comment by quchenhao** [comment]
> @tammybailey, I'm probably missing something obvious, but I want to be sure: I don't see anywhere we check an import load for `compare_timestamp=(oldest_timestamp)`, and I don't see a test path for it. Is `oldest_timestamp` the default or something so it's not necessary?

Yes, the oldest is the default. So it is compatible to the old mondodb usage that doesn't set this value.

**Comment by agorrod** [comment]
This change looks good to me. We've encountered a few cases in the past where changing configuration options hurts upgrade/downgrade. The changes here seem to be transient (nothing saved to disk), and backward compatible. So I think they are safe, but I'd appreciate @keithbostic if you can think about that before merging.

**Review by agorrod — APPROVED** [review]

**Review by tammybailey — APPROVED** [review]
lgtm

File: dist/api_data.py [hunk 1]
```diff
@@ -1938,19 +1939,19 @@ def __ge__(self, other):
...
+        recent \c oldest_timestamp set with WT_CONNECTION::set_timestamp, \c oldest_reader
+        returns the minimum of the read timestamps of all active readers \c pinned returns
+        the minimum of the \c oldest_timestamp and the read timestamps of all active readers,
+        \c recovery returns the timestamp of the most recent stable checkpoint taken prior to a
+        shutdown and \c stable_timestamp returns the most recent \c stable_timestamp set with
+        WT_CONNECTION::set_timestamp. (The \c oldest and \c stable arguments are deprecated
+        short-hand for \c oldest_timestamp and \c stable_timestamp, respectively.) See @ref
+        transaction_timestamps''',
+        choices=['all_durable','last_checkpoint','oldest',
+            'oldest_timestamp','oldest_reader','pinned','recovery','stable','stable_timestamp']),
```
Comment (sueloverso) [inline dist/api_data.py:1954]: Alphabetize with oldest_reader.

File: src/txn/txn_timestamp.c [hunk 2]
```diff
@@ -191,7 +191,8 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
         /* Read-only value forever. Make sure we don't used a cached version. */
         WT_BARRIER();
         ts = txn_global->last_ckpt_timestamp;
-    } else if (WT_STRING_MATCH("oldest", cval.str, cval.len)) {
+    } else if (WT_STRING_MATCH("oldest_timestamp", cval.str, cval.len) ||
+      WT_STRING_MATCH("oldest", cval.str, cval.len)) {
```
Comment (sueloverso) [inline src/txn/txn_timestamp.c:195]: Alphabetize oldest before oldest_timestamp.

File: src/txn/txn_timestamp.c [hunk 3]
```diff
@@ -203,7 +204,8 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
     else if (WT_STRING_MATCH("recovery", cval.str, cval.len))
         /* Read-only value forever. No lock needed. */
         ts = txn_global->recovery_timestamp;
-    else if (WT_STRING_MATCH("stable", cval.str, cval.len)) {
+    else if (WT_STRING_MATCH("stable_timestamp", cval.str, cval.len) ||
+      WT_STRING_MATCH("stable", cval.str, cval.len)) {
```
Comment (sueloverso) [inline src/txn/txn_timestamp.c:208]: same here.

**Review by sueloverso — APPROVED** [review]
LGTM with a few minor nits. Is there any plan to get the server code to use the new API and deprecate and remove the old ones? Should there be a SERVER ticket for that and then a new WT ticket to remove it? Upgrade/downgrade should not matter because this is not stored anywhere, it is just calling code.

**Comment by keithbostic** [comment]
> We've encountered a few cases in the past where changing configuration options hurts upgrade/downgrade. The changes here seem to be transient (nothing saved to disk), and backward compatible. So I think they are safe, but I'd appreciate @keithbostic if you can think about that before merging.

I don't anticipate any problems, but I'll run a MongoDB patch because I've been wrong before.

File: dist/api_data.py (hunk 1 above)
Comment (keithbostic) [reply dist/api_data.py]: Pushed in 667959b6b.

File: src/txn/txn_timestamp.c (hunk 2 above)
Comment (keithbostic) [reply src/txn/txn_timestamp.c:195]: Hopefully the "standard" one will match first, so we'll avoid a function call.

File: src/txn/txn_timestamp.c (hunk 3 above)
Comment (keithbostic) [reply src/txn/txn_timestamp.c:208]: Hopefully the "standard" one will match first, so we'll avoid a function call.

**Comment by keithbostic** [comment]
> Is there any plan to get the server code to use the new API and deprecate and remove the old ones?

There is no such plan.

**Comment by keithbostic** [comment]
MDB patch build: https://evergreen.mongodb.com/version/61d8bf18a4cf47760367a85b?redirect_spruce_users=true

**Pull request merged**